src/hera/workflows/script.py

```diff
@@ -232,16 +236,34 @@ def _build_script(self) -> _ModelScriptTemplate:
 def _get_parameters_from_callable(source: Callable) -> Optional[List[Parameter]]:
     # If there are any kwargs arguments associated with the function signature,
     # we store these as we can set them as default values for argo arguments
-    source_signature: Dict[str, Optional[object]] = {}
+    #
+    # Take all the annotation information and make sure it is stored.
+    parameters = []
+
     for p in inspect.signature(source).parameters.values():
         if p.default != inspect.Parameter.empty and p.kind == inspect.Parameter.POSITIONAL_OR_KEYWORD:
-            source_signature[p.name] = p.default
+            default = p.default
         else:
-            source_signature[p.name] = MISSING
+            default = MISSING
+
+        param = Parameter(name=p.name, default=default)
 
-    if len(source_signature) == 0:
+        # add docs for this feature in docs/getting-started/walk-through/advanced-hera-features.md
+        if global_config.experimental_features["script_annotations"]:
+            if get_origin(p.annotation) is Annotated and isinstance(get_args(p.annotation)[1], Parameter):
+                annotation = get_args(p.annotation)[1]
+                for attr in ["enum", "description", "default", "name"]:
+                    if hasattr(annotation, attr) and getattr(annotation, attr) is not None:
+                        if attr == "default" and getattr(param, attr) is not None:
```

Comment:
We can stick to the direct access for this special case (easier to understand)
```suggestion
                        if attr == "default" and param.default is not None:
```